core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -470,7 +469,7 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
           nonExistentTopics.foreach(topic => zkUtils.deletePathRecursive(getDeleteTopicPath(topic)))
         }
         topicsToBeDeleted --= nonExistentTopics
-        if(topicsToBeDeleted.size > 0) {
+        if(controller.config.deleteTopicEnable && topicsToBeDeleted.size > 0) {
```

Comment:
The previous logs are at debug level and warn (in case of non-existent topics). IMO, its ok to put the check here. Since its a simple check having a separate class would be an overkill for this, I think.